npm/qsharp/src/language-service/language-service.ts

```diff
@@ -51,7 +51,12 @@ export type LanguageServiceEvent =
 // for running the compiler in the same thread the result will be synchronous (a resolved promise).
 export interface ILanguageService {
   updateConfiguration(config: IWorkspaceConfiguration): Promise<void>;
-  updateDocument(uri: string, version: number, code: string): Promise<void>;
+  updateDocument(
+    uri: string,
+    version: number,
+    code: string,
+    languageId: string,
```

Comment:
not an expert here, but I think this is the spot where you'd want to make `languageId` optional with a default of `"qsharp"` to avoid having it be a breaking change for consumers (the fact that playground had to be updated is usually a good hint it would be a breaking change for QCOM).

Reply:
That was my thought too, was hoping @minestarks or @billti could comment.

Reply:
Look at getCircuit on ICompiler (and get_circuit in wasm/src/lib.rs) where the operation is optional and has get a default if not specified. Does that approach work?